src/topp/IDFilter.cpp

```diff
@@ -227,19 +229,16 @@ class TOPPIDFilter :
                                  "Similar to n_peptide_hits=1, but if there are ties between two or more highest scoring hits, none are kept.");
     registerStringOption_("best:n_to_m_peptide_hits", "[min]:[max]", ":", "Peptide hit rank range to extracts", false, true);
 
-    registerStringOption_("length", "[min]:[max]", ":", "Keep only peptide hits with a sequence length in this range.", false);
-
-    registerStringOption_("charge", "[min]:[max]", ":", "Keep only peptide hits with charge states in this range.", false);
 
     registerFlag_("var_mods", "Keep only peptide hits with variable modifications (as defined in the 'SearchParameters' section of the input file).", false);
 
-    registerFlag_("unique", "If a peptide hit occurs more than once per peptide ID, only one instance is kept.");
+    registerFlag_("unique_per_spectrum", "If a peptide-spectrum match occurs more than once for a spectrum, only one instance is kept.", true);
     registerFlag_("unique_per_protein", "Only peptides matching exactly one protein are kept. Remember that isoforms count as different proteins!");
```

Comment:
I am not a big fan of unique_per_protein, too.
Wouldn't "remove_shared_peptides" or "remove_ambiguous_peptides" be easier to understand?

Reply:
"remove_ambiguous_peptides" - this maybe.

But "remove_shared_peptides" is usually called "unique" - I am not sure what would be the best option.

Reply:
Because unique_per_protein (especially with the meaning of the option right above it) makes me definitely think about removing all but one peptide per protein.

Reply:
agree with Julianus.
What is the problem with remove_shared_peptides - seems best fit to me?

Reply:
There is no problem per se - we can go with it. The only thing I see is that in a lot of other software it is just called "unique" and this may confuse some users.

Reply:
Yes, I see but this is dangerous if the user opens an old workflow where "unique" had a different meaning. Don't know if that will be an issue but if it is, it will be super hard to debug for users.

Reply:
Ok, then we should go with the suggestion of Julianus.